src/macros.h

```diff
@@ -143,6 +181,69 @@
 #define SIZE_OF(typ, id) sizeof(((typ *)0)->id)
 #endif // !SIZE_OF
 
+ /**
+ * In order to determine the offset of an element of a structure, use the
+ * following macro as shown:
+ * 
+ * Example usage:
+ *  typedef struct _ABC {
+ *      sint32    A;
+ *      sint32    B;
+ *      sint16    C;
+ * } ABC, * PTR_ABC
+ * 
+ * OFFSET_OF(struct _ABC, C)
+ */
+#ifndef OFFSET_OF
+#define OFFSET_OF(typ, id) (unsigned int)&(((typ*)0)->id)
+#endif // !OFFSET_OF
+
+ /**
+ * Determines the alignment of a type.
+ * 
+ * Example usage:
+ *  size_t alignment = ALIGN_OF(int);
+ */
+#ifndef ALIGN_OF
+#if defined(__MWERKS__) || defined(__CWCC__) // Metrowerks compilers exist for many platforms.
+#define ALIGN_OF(type) ((size_t)__alignof__(type))
+#elif !defined(COMPILER_GNUC) || (__GNUC__ >= 3) // GCC 2.x doesn't do __alignof correctly all the time.
+#define ALIGN_OF __alignof
+#else
+#define ALIGN_OF(type) ((size_t)OFFSET_OF(struct{ char c; type m; }, m))
+#endif // Metrowerks
+#endif // !ALIGN_OF
+
+/**
+ * Safely delete an object and set the pointer to NULL.
+ * 
+ * Example usage:
+ *  SAFE_DELETE(MyPtr);
+ */
+#ifndef SAFE_DELETE
```

Comment:
Calling delete on a nullptr is fine and without that I don't really see the point in using macros like these which is why I took them out in thyme and chronoshift in the first place because we weren't using them and were never likely to. They just obfuscate the code to someone else reading it who seeing it used might think "Hmm, what is a "safe" delete?" while there is no confusion if they see a delete followed by a nullptr assignment.